repo/db/chat.go

```diff
@@ -150,44 +150,63 @@ func (c *ChatDB) GetMessages(peerID string, subject string, offsetId string, lim
 	return ret
 }
 
-func (c *ChatDB) MarkAsRead(peerID string, subject string, outgoing bool, messageId string) (string, error) {
+func (c *ChatDB) MarkAsRead(peerID string, subject string, outgoing bool, messageId string) (string, bool, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
+	updated := false
 	outgoingInt := 0
 	if outgoing {
 		outgoingInt = 1
 	}
 	tx, err := c.db.Begin()
 	if err != nil {
-		return "", err
+		return "", updated, err
 	}
 	var stmt *sql.Stmt
 	if messageId != "" {
 		stmt, _ = tx.Prepare("update chat set read=1 where peerID=? and subject=? and outgoing=? and timestamp<=(select timestamp from chat where messageID=?)")
+		stm := "select messageID from chat where peerID=? and subject=? and outgoing=? and read=0 and timestamp<=(select timestamp from chat where messageID=?)"
```

Comment:
I would personally so an EXISTS query and scan the result  into `&updated, which is clearer IMO but a very tiny detail and not a problem.